db/test/fns/queue_test.js

```diff
@@ -1044,42 +1044,67 @@ suite(testing.suiteName(), function() {
     const expires = taskcluster.fromNow('2 hours');
     const firstClaim = taskcluster.fromNow('0 hours');
     const create = async (db, options = {}) => {
-      await db.fns.create_queue_worker_tqid(
-        options.taskQueueId || 'prov/wt',
-        options.workerGroup || 'wg',
-        options.workerId || 'wi',
-        options.quarantineUntil || quarantineUntil,
-        options.expires || expires,
-        options.firstClaim || firstClaim,
-        options.recentTasks || JSON.stringify([{
-          recent: "task",
-        }]),
-      );
+      // we don't have a "create" function anymore, so we emulate it
+      await db.fns.queue_worker_seen({
+        task_queue_id_in: options.taskQueueId || 'prov/wt',
+        worker_group_in: options.workerGroup || 'wg',
+        worker_id_in: options.workerId || 'wi',
+        expires_in: options.expires || expires,
+      });
+
+      if (options.quarantineUntil) {
+        await db.fns.quarantine_queue_worker({
+          task_queue_id_in: options.taskQueueId || 'prov/wt',
+          worker_group_in: options.workerGroup || 'wg',
+          worker_id_in: options.workerId || 'wi',
+          quarantine_until_in: options.quarantineUntil,
+        });
+      }
+
+      for (let task of options.recentTasks || [{ taskId: 'recent', runId: 0 }]) {
+        await db.fns.queue_worker_task_seen({
+          task_queue_id_in: options.taskQueueId || 'prov/wt',
+          worker_group_in: options.workerGroup || 'wg',
+          worker_id_in: options.workerId || 'wi',
+          task_run_in: task,
+        });
+      }
     };
 
     helper.dbTest('no such queue worker', async function(db) {
       const res = await db.fns.get_queue_worker_tqid('prov/wt', 'wg', 'wi', new Date());
       assert.deepEqual(res, []);
     });
 
-    helper.dbTest('create_queue_worker_tqid / get_queue_worker_tqid', async function(db) {
-      await create(db);
+    helper.dbTest('create_queue_worker_tqid (deprecated) / get_queue_worker_tqid', async function(db) {
+      await db.deprecatedFns.create_queue_worker_tqid(
+        'prov/wt',
+        'wg',
+        'wi',
+        quarantineUntil,
+        expires,
+        firstClaim,
+        JSON.stringify([{ taskId: 'recent', runId: 0 }]),
+      );
       const res = await db.fns.get_queue_worker_tqid('prov/wt', 'wg', 'wi', new Date());
       assert.equal(res[0].task_queue_id, 'prov/wt');
       assert.equal(res[0].worker_group, 'wg');
       assert.deepEqual(res[0].quarantine_until, quarantineUntil);
       assert.deepEqual(res[0].expires, expires);
       assert.deepEqual(res[0].first_claim, firstClaim);
-      assert.deepEqual(res[0].recent_tasks, [{ recent: "task" }]);
+      assert.deepEqual(res[0].recent_tasks, [{ taskId: 'recent', runId: 0 }]);
```

Comment:
Is the change in the value of ``.recent_tasks`` going to cause an issue when reading existing data?

Reply:
The old value was just used in the tests; the `{taskId, runId}` format is what's used in production.  At one point in development of this PR I was passing `task_id_in` and `run_id_in` to the DB function, so I had to change this test data to match the correct format.  I subsequently changed that (to just passing `task_in`), but having test data in an accurate format is still a good idea.